src/pkg/caendr/caendr/api/strain.py

```diff
@@ -228,22 +228,20 @@ def generate_bam_bai_download_script(species, release, signed=False):
   # Add download statements for each strain
   for strain in strain_listing:
     yield f'# Strain: {strain}\n'
-    bam_path = f'{bam_prefix}/{strain}.bam'
-    bai_path = f'{bam_prefix}/{strain}.bam.bai'
+
+    # Generate filenames
+    bam_fname = f'{strain}.bam'
+    bai_fname = f'{strain}.bam.bai'
 
     # Generate download URLs
-    if signed:
-      bam_url = generate_download_signed_url_v4(bucket_name, bam_path, expiration=expiration, credentials=credentials)
-      bai_url = generate_download_signed_url_v4(bucket_name, bai_path, expiration=expiration, credentials=credentials)
-    else:
-      bam_url = generate_blob_url(bucket_name, bam_path)
-      bai_url = generate_blob_url(bucket_name, bai_path)
+    bam_url = generate_blob_uri(bucket_name, bam_prefix, bam_fname, **sign_dict)

```

Comment:
New unified `generate_blob_uri` functions takes the path as a variadic list argument, so we can insert the `bam_prefix` here and let the function handle joining the paths (rather than assuming a `/` format here)

Reply:
It also handles signing based on the `schema`, which is passed in as part of `sign_dict`